octoprint_slic3r/__init__.py

```diff
@@ -321,6 +322,7 @@ def do_slice(self, model_path, printer_profile, machinecode_path=None, profile_p
 						self._slic3r_logger.debug("stdout: " + stdout_line.strip())
 					if stderr_line:
 						self._slic3r_logger.debug("stderr: " + stderr_line.strip())
+                                                if ( len(stderr_line.strip()) > 0 ): last_error = stderr_line.strip()
```

Comment:
I think that you can remove the if and simply always assign.  Because if the length is not greater than zero, it will just assign `""`, which is the current value anyway.

So you can just write:

```
last_error = stderr_line.strip()
```

Reply:
I tested it and that's exactly where the `if` came in :)
the sterr log seems to contain one empty line after the error message, this avoids overwriting the actual message with an empty line

Reply:
I don't understand.  Either `stderr_line.strip()` is length 0 or it's greater, right?

If it's greater, then you write it into last_error.

If it's zero then it's the same as `""`, which is what is already in last_error, so writing it to last_error makes no difference.

So couldn't you just assign it to last_error either way?

Reply:
in the second case, `length == 0` the assumption is different: from a previous run of the loop the `last_error` already contains a meaningful message. Now, without the `if` the message in `last_error` will be overwritten by `""`.

Reply:
Ah, I didn't see that it's in a loop.  Okay!